prime-router/src/main/kotlin/encoding/FHIR.kt

```diff
@@ -0,0 +1,74 @@
+package gov.cdc.prime.router.encoding
+
+import ca.uhn.hl7v2.model.Message
+import ca.uhn.hl7v2.model.v251.segment.MSH
+import io.github.linuxforhealth.fhir.FHIRContext
+import io.github.linuxforhealth.hl7.ConverterOptions
+import io.github.linuxforhealth.hl7.message.HL7MessageEngine
+import io.github.linuxforhealth.hl7.resource.ResourceReader
+import org.apache.logging.log4j.kotlin.Logging
+import org.hl7.fhir.r4.model.Bundle
+
+fun Bundle.encode(): String {
+    return FHIR.encode(this)
+}
+
+class FHIR {
+    companion object : Logging {
+        val resourceReader = ResourceReader.getInstance()
+
+        val messagetemplates = resourceReader.getMessageTemplates()
+
+        val engine = getMessageEngine()
+
```

Comment:
The way this is implemented it's not really possible to `spyk` or `mockk` the ResourceReader. It would be better to pass in a function to the class that creates your ResourceReader or lets someone pass in a mocked one, OR, don't make this as a singleton and instead convert these public var's that can be overwritten.

My concern is that there could be all kinds of underlying logic in ResourceReader that could screw with tests, or just not work and we'll have to do some rework to address this later.

Reply:
This is only used to generate the messageTemplates. I'll pull those into something that can be passed in.